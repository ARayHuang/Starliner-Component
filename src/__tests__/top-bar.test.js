import React from 'react';
import { shallow, mount } from 'enzyme';
import TopBar from '../components/top-bar';

describe('TopBar', () => {
	it('handle default props', () => {
		const  {
			hasDivider,
			userProfile,
			onLogout
		} = TopBar.defaultProps;

		expect(hasDivider).toBe(false);
		expect(userProfile).toEqual({});
		expect(onLogout).toBeInstanceOf(Function);
	});

	it('should render correctly', () => {
		const props = {
			userProfile: {
				username: 'mock-username',
				userEmail: 'mock-email'
			},
			hasDivider: true,
			childrenSx: {
				color: '#3f4a57'
			},
			userProfileSx: {
				backgroundColor: '#ebf2fa'
			},
			onLogout: () => {}
		};

		const wrapper = shallow(
			<TopBar {...props}>
				<div>mock-children</div>
			</TopBar>
		)

		expect(wrapper).toMatchSnapshot();
	})

	it('should mount in a full DOM', () => {
		const props = {
			userProfile: {
				username: 'mock-username',
				userEmail: 'mock-email'
			},
			hasDivider: true,
			childrenSx: {
				color: '#3f4a57'
			},
			userProfileSx: {
				backgroundColor: '#ebf2fa'
			},
			onLogout: () => {}
		};

		const wrapper = mount(
			<TopBar {...props} />
		);

		expect(wrapper.props().userProfile).toEqual(props.userProfile);
		expect(wrapper.props().hasDivider).toEqual(props.hasDivider);
		expect(wrapper.props().childrenSx).toEqual(props.childrenSx);
		expect(wrapper.props().userProfileSx).toEqual(props.userProfileSx);
		expect(wrapper.props().onLogout).toEqual(props.onLogout);
	})
})
