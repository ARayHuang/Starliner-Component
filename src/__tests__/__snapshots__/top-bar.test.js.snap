// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render correctly 1`] = `
<Fragment>
  <Stack
    h="50px"
  >
    <Flex
      alignItems="center"
    >
      <Box
        p="2"
        sx={
          Object {
            "color": "#3f4a57",
          }
        }
      >
        <div>
          mock-children
        </div>
      </Box>
      <Spacer />
      <Box
        p="2"
        sx={
          Object {
            "backgroundColor": "#ebf2fa",
          }
        }
      >
        <span>
          mock-username
          /
        </span>
        <span
          style={
            Object {
              "color": "#bdcfe7",
              "fontSize": "7pt",
            }
          }
        >
          mock-email
        </span>
        <Menu>
          <MenuButton>
            <Avatar
              name="mock-username"
              size="sm"
            />
          </MenuButton>
          <MenuList>
            <MenuItem
              classnames="logout-menu"
              onClick={[Function]}
            >
              Logout
            </MenuItem>
          </MenuList>
        </Menu>
      </Box>
    </Flex>
  </Stack>
  <Divider />
</Fragment>
`;
